lib/vrl/tests/tests/diagnostics/unnest_typedefs.vrl

```diff
@@ -0,0 +1,15 @@
+# result:
```

Comment:
Do we already have an integration test that validates what happens if you assign an array to `.`? I know it's somewhat out-of-scope for this PR, but I think it's worth adding if it doesn't exist yet.

Reply:
We do have this one at least:

https://github.com/timberio/vector/blob/3629646513f17bb4328cc3b92ee5aa35f0b4f38c/tests/behavior/transforms/remap.toml#L1-L19

Reply:
Ah, yeah! I forgot how `. = <array>` worked, but in the case of VRL it just sets the target object to an array, and that's it, whereas the `remap` transform does the actual work of turning an array into a list of multiple events.

Gotcha, in that case there's nothing to test on VRL's side itself.